### PR TITLE
add clusterrouter, first time

### PR DESCRIPTION
**Pull request by xyz2277**

### CODE REVIEW

File: cmd/knode-manager/main.go [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+)
+
+func main() {
+	ctx, cancel := context.WithCancel(context.Background())
+	sig := make(chan os.Signal, 1)
+	signal.Notify(sig, syscall.SIGINT, syscall.SIGTERM)
+	go func() {
+		<-sig
+		cancel()
+	}()
```
Comment (wuyingjun-lucky) [inline cmd/knode-manager/main.go]: what does `sig` for

Comment (xyz2277) [reply cmd/knode-manager/main.go]: > what does `sig` for

For keeping the main thread from exiting.

Comment (wuyingjun-lucky) [reply cmd/knode-manager/main.go]: ctx := apiserver.SetupSignalContext()
what about using this ？

Comment (xyz2277) [reply cmd/knode-manager/main.go]: it's a better way

**Pull request merged**